src/components/ui/WithTooltip/index.tsx

```diff
@@ -55,7 +61,19 @@ export const WithTooltip = <P extends object>(
     return () => document.removeEventListener('scroll', hideToolTip, true);
   }, []);
 
-  const container = document.getElementById(container_id);
+  useEffect(() => {
+    const container = document.getElementById(container_id || 'Tooltip__container');
+
+    if (!container) {
+      console.warn(`
+        Attempted to use 'WithTooltip' but could not find container element.
+        Pass a valid element ID or add 'Tooltip__container' ID to existing element
+      `);
+      return;
+    }
+
+    setContainer(container)
+  }, []);
```

Comment:
This `useEffect` needs to be dependent on the `container_id` prop right? Like if `container_id` changes, the component should get re-rendered.

Reply:
Yep! Thanks for catching that